### PR TITLE
pushup: init at 0.2

### PR DESCRIPTION
**Pull request by paulsmith**
## Description of changes

[Pushup](https://pushup.adhoc.dev/) is a new web framework tool for making page-oriented web apps using the Go programming language.

Pushup compiles a novel syntax that combines Go and HTML down to pure Go code. The Pushup binary builds self-contained, standalone web app executables. It features file-based routing and support for modern HTML-over-the-wire projects like HTMX.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rjpcasalino** [comment]
Result of `nixpkgs-review pr 276158` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pushup</li>
  </ul>
</details>

```
[nix-shell:~/.cache/nixpkgs-review/pr-276158-2/results/pushup/bin]$ ./pushup --version
Pushup 0.0.1
```

**Review by rjpcasalino — APPROVED** [review]

File: maintainers/maintainer-list.nix
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:14129]: please add either email and/or matrix, so other maintainer are able to contact you directly.
And make it it's own commit called `maintainers: add paulsmith`

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:30]: this should be fine to have in one line

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:26]:
```suggestion
    description = "A web framework for Go";
```
We don't want the package name at the start of the description.

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:21]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
  allowGoReference = true;
  # add a comment about needing go at runtime
  postInstall = ''
    wrapProgram $out/bin/${pname} --prefix PATH : ${
      lib.makeBinPath [ go ]
```
instead of passing pkgs we pass the attributes directly and callPackage will do the rest.

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:1]:
```suggestion
{ fetchFromGitHub, buildGoModule, lib, go, makeWrapper }:
```

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:15]: what is CGO used for/why are we disabling it?

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix
Comment (paulsmith) [reply maintainers/maintainer-list.nix]: Done. https://github.com/NixOS/nixpkgs/pull/276197

File: pkgs/by-name/pu/pushup/package.nix
Comment (paulsmith) [reply pkgs/by-name/pu/pushup/package.nix]: Pushup does not use CGO so it's common to disable it explicitly for builds.

File: pkgs/by-name/pu/pushup/package.nix
Comment (paulsmith) [reply pkgs/by-name/pu/pushup/package.nix]: @Janik-Haag Thank you for your review of my PR. I believe I have addressed your requests in the latest commits,

**Comment by Janik-Haag** [comment]
You'll have to rebase on master so ofborg can find your maintainer object that we just added.
Also please squash the commits so we just have `pushup: init at 0.2`, we do this because nixpkgs tries to have a atomic git commit history.

**Review by Janik-Haag — APPROVED** [review]

**Comment by Janik-Haag** [comment]
@ofborg build pushup

**Comment by Janik-Haag** [comment]
just waiting for ci now.

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:8]:
```suggestion
    repo = "pushup";
```

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:2]:
```suggestion
{ fetchFromGitHub, buildGoModule, lib, go, makeWrapper }:

buildGoModule rec {
```

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:15]:
```suggestion

```

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:19]:
```suggestion

```

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:32]:
```suggestion
    mainProgram = "pushup";
```

File: pkgs/by-name/pu/pushup/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/pu/pushup/package.nix:31]:
```suggestion
    changelog = "https://github.com/adhocteam/pushup/blob/${src.rev}/CHANGELOG.md";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by Janik-Haag** [comment]
@paulsmith please change the pname to the static value as torres suggested, but leave the comments there.
This was a slight oversight on my part and having it set to pname can have bad interactions with overrides and inherits.

@AndersonTorres next time you do a review please write at least some explanation on why you are suggesting some changes.

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [reply pkgs/by-name/pu/pushup/package.nix:2]: this is a formatting nit and for me it's fine either ways, we will have a official for-matter soon :tm:

File: pkgs/by-name/pu/pushup/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/pu/pushup/package.nix:22]:
```suggestion
    wrapProgram $out/bin/${meta.mainProgram} --prefix PATH : ${
```
for completeness sake

**Comment by rjpcasalino** [comment]
@Janik-Haag 
>having it set to pname can have bad interactions with overrides and inherits

what's an example of a bad interaction for my own education?

**Review by rjpcasalino — APPROVED** [review]

**Comment by AndersonTorres** [comment]
> what's an example of a bad interaction for my own education? 

An outdated example: the Python Imaging Library has `pname = pil` but `repo = Imaging`.

**Comment by AndersonTorres** [comment]
> @AndersonTorres next time you do a review please write at least some explanation on why you are suggesting which change.  

The comments convey no new information.

**Comment by rjpcasalino** [comment]
I agree w/ @AndersonTorres about these comments but isn't a showstopper for me

**Comment by Janik-Haag** [comment]
I looked at some other go packages before hand and most of them have comments about that so I would prefer to leave them in.

**Comment by paulsmith** [comment]
@rjpcasalino @Janik-Haag @AndersonTorres Thank you, appreciate you all reviewing this PR and helping me get my first Nix contribution merged!

**Pull request merged**
